experimental/unitsconversion/src/measureunit.rs

```diff
@@ -3,75 +3,75 @@
 /// A parser for the CLDR unit identifier (e.g. `meter-per-square-second`)
 pub struct MeasureUnitParser<'data> {
     /// Contains the payload.
-    payload: &'data ZeroTrie<ZeroVec<'data, u8>>,
+    payload: &'data ZeroTrieSimpleAscii<ZeroVec<'data, u8>>,
 }
 
 impl<'data> MeasureUnitParser<'data> {
     // TODO: revisit the public nature of the API. Maybe we should make it private and add a function to create it from a ConverterFactory.
     /// Creates a new MeasureUnitParser from a ZeroTrie payload.
     #[cfg(feature = "datagen")]
-    pub fn from_payload(payload: &'data ZeroTrie<ZeroVec<'data, u8>>) -> Self {
+    pub fn from_payload(payload: &'data ZeroTrieSimpleAscii<ZeroVec<u8>>) -> Self {
         Self { payload }
     }
 
-    // TODO: complete all the cases for the prefixes.
-    // TODO: consider using a trie for the prefixes.
-    /// Extracts the SI prefix.
+    /// Get the unit id.
     /// NOTE:
-    ///    if the prefix is found, the function will return (SiPrefix, part without the prefix string).
-    ///    if the prefix is not found, the function will return (SiPrefix { power: 0, base: Base::Decimal }, part).
-    fn get_si_prefix(part: &str) -> (SiPrefix, &str) {
-        let (si_prefix_base_10, part) = get_si_prefix_base_ten(part);
-        if si_prefix_base_10 != 0 {
-            return (
-                SiPrefix {
-                    power: si_prefix_base_10,
-                    base: Base::Decimal,
-                },
-                part,
-            );
+    ///    if the unit id is found, the function will return (unit id, part without the unit id and without `-` at the beginning of the remaining part if it exists).
+    ///    if the unit id is not found, the function will return None.
+    fn get_unit_id<'a>(&'a self, part: &'a str) -> Result<(u16, &str), ConversionError> {
+        let mut cursor = self.payload.cursor();
+        let mut longest_match = Err(ConversionError::InvalidUnit);
+
+        for (i, byte) in part.bytes().enumerate() {
+            cursor.step(byte);
+            if cursor.is_empty() {
+                break;
+            }
+            if let Some(value) = cursor.take_value() {
+                longest_match = Ok((value as u16, &part[i + 1..]));
+            }
+        }
+        longest_match
+    }
+
+    fn get_power<'a>(&'a self, part: &'a str) -> Result<(u8, &str), ConversionError> {
+        let (power, part_without_power) = get_power(part);
+        if part_without_power.len() == part.len() {
+            return Ok((power, part));
         }
 
-        let (si_prefix_base_2, part) = get_si_prefix_base_two(part);
-        if si_prefix_base_2 != 0 {
-            return (
-                SiPrefix {
-                    power: si_prefix_base_2,
-                    base: Base::Binary,
-                },
-                part,
-            );
+        if part_without_power.starts_with("-") {
+            return Ok((power, &part_without_power[1..]));
         }
```

Comment:
Suggestion (optional): you can avoid the indexing by using `split_first`, like

```rust
if let Some((&'-', remainder)) = part_without_power.split_first() {
    return Ok((power, remainder));
}
```

However, I think this only works if you deal with `[u8]` instead of `str`.

You have a lot of indexing here but it is all guarded so it's fine. You'll need to add some clippy suppressions, though.

Reply:
done